services/app-api/utils/formTemplates/formTemplates.ts

```diff
@@ -92,10 +116,12 @@ export async function getOrCreateFormTemplate(
       throw err;
     }
 
+    const newestTemplateMetadata = await getNewestTemplateVersion(reportType);
+
     // If we didn't find any form templates, start version at 1.
     const newFormTemplateVersionItem: FormTemplate = {
-      versionNumber: mostRecentTemplateVersion?.versionNumber
-        ? (mostRecentTemplateVersion.versionNumber += 1)
+      versionNumber: newestTemplateMetadata?.versionNumber
+        ? (newestTemplateMetadata.versionNumber += 1)
```

Comment:
we still version off of the latest for simplicity. So whether you create a MCPAR or a PCCM, if it's new it will get a +1 to the latest version and save in the same table/bucket

Reply:
To be clear, since PCCM form template versions are stored with reportType = MCPAR, we will have  MCPAR version 17 (say) be a full MCPAR, and MCPAR version 18 be PCCM, and MCPAR version 19 be a full MCPAR again (with some differences compared to v17, such that the hash no longer matches). The only way for a developer browsing the form template version table to know whether a given entry for the MCPAR reportType is PCCM will be for them to pull the form template from S3.

I don't anticipate ever needing or wanting to do so myself, so I have no problem with that.

I also am not sure that this comment adds any clarity. Um, sorry. The code is good though.

Reply:
Thanks for clarifying, Ben! That helps.

Yes if we needed to find a PCCM template we could always look in the report table first, search for a report that selected "Yes" to PCCM, get the templateId from there, etc.